src/Microsoft.Extensions.DependencyModel/DependencyContextWriter.cs

```diff
@@ -75,11 +102,22 @@ private JObject WriteCompilationOptions(CompilationOptions compilationOptions)
             {
                 o[DependencyContextStrings.EmitEntryPointPropertyName] = compilationOptions.EmitEntryPoint;
             }
+            if (compilationOptions.GenerateXmlDocumentation != null)
```

Comment:
These seem ripe for refactoring into a helper :). `AddProperty(jobj, name, value)`